### DISCUSSION TITLE
How to Install ImageMagick in UBI9 container

### DISCUSSION THREAD

**Discussion by NadunW** [ImageMagick/ImageMagick#7486] [Help]
### ImageMagick version

7

### Operating system, version and so on

UBI9

### Description

Hi,

I was trying to install IM in UBI9 container, but it failed to find dependencies. 

Highly appreciate any guide or tip.

**Comment by icsy7867** [comment]
This was working through EPEL, but it stopped for me due to dependencies.  I also tried adding the REMI repo.  But no luck.

So then I tried to build a container that built it from source:
https://imagemagick.org/script/install-source.php
https://linuxcapable.com/how-to-install-imagemagick-on-fedora-linux/

```
# dnf install gcc clang make git -y
# git clone --depth 1 --branch 7.1.1-43 https://github.com/ImageMagick/ImageMagick.git ImageMagick-7.1.1
# cd ImageMagick-7.1.1
# ./configure
# make
# make install
```

But the dependencies add almost a gigabyte to the image.  So I tried using a builder stage in my docker file and copied over the files into my new container:
```
/usr/local/include/ImageMagick-7
/usr/local/share/doc/ImageMagick-7
/usr/local/bin/
/usr/local/lib/
```
But I was still getting issues with my JPG image and it not having a definition for it.  So I am missing some directories somewhere....

Ultimately I gave up, and moved to a debian based image, where an:

```
apt update && apt install imagemagick -y
```
worked without issue...
